### ISSUE TITLE
Error: _iterate function includes custom methods of Array's prototype chain

### ISSUE THREAD

**Issue by LostCrew** [tombatossals/angular-leaflet-directive#899]
Since I have my `Array.prototype.find` polyfill for non-modern browser, an error occurs while adding mapEvents with addEvents call. The function is passed to Leaflet's `addEventListener` mixin, which accepts only strings.

Loop in `_iterate`:

![screen shot 2015-08-25 at 16 45 08](https://cloud.githubusercontent.com/assets/1306576/9469884/a4417538-4b49-11e5-96fe-daaaa5152865.png)

`eventName` is function, not string:

![screen shot 2015-08-25 at 16 40 14](https://cloud.githubusercontent.com/assets/1306576/9469856/81f4776e-4b49-11e5-85ea-1263097e42de.png)

Final error:

![screen shot 2015-08-25 at 16 44 24](https://cloud.githubusercontent.com/assets/1306576/9469941/dec1b77c-4b49-11e5-86ef-2445a4462f80.png)

Solution: https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Statements/for...in#Iterating_over_own_properties_only

**Comment by nmccready** [comment]
Ok can you make a PR which makes it better! Thanks

**Comment by nmccready** [comment]
@LostCrew I'll take care of this thanks.

**Comment by nmccready** [comment]
haha looks like this is already done on master.. need to find the dupe issue
https://github.com/tombatossals/angular-leaflet-directive/blob/master/src/services/leafletIterators.js#L146

**Comment by nmccready** [comment]
#884

**Comment by LostCrew** [comment]
@nmccready perfect, can you estimate when you're going to make a release?

**Comment by nmccready** [comment]
@LostCrew pending on #874 as I need to know what version number to release.
